### PR TITLE
authentication for users for votes and reviews

### PR DESCRIPTION
**Pull request by Cheese-Baron**

### CODE REVIEW

File: app/controllers/api/v1/amusement_parks_controller.rb [hunk 1]
```diff
@@ -6,11 +6,18 @@ def index
...
+      admin_status = current_user.admin?
+    end
+
     render json: {
       amusement_park: AmusementPark.find(params[:id]),
       reviews: serialized_review,
       rides: serialized_rides,
-      current_user_id: current_user_id
+      current_user_id: current_user_id,
+      adminStatus: admin_status
```
Comment (bkincart) [inline app/controllers/api/v1/amusement_parks_controller.rb]: this should also be snake_case :)

File: app/controllers/api/v1/reviews_controller.rb [hunk 2]
```diff
@@ -19,7 +21,18 @@ def show
     likes = review.tally_likes
     dislikes = review.tally_dislikes
 
-    render json: { likes: likes, dislikes: dislikes }
+    render json: { likes: likes, dislikes: dislikes, adminStatus: admin_status? }
```
Comment (bkincart) [inline app/controllers/api/v1/reviews_controller.rb]: This should be snake_case too!

File: app/controllers/api/v1/reviews_controller.rb [hunk 3]
```diff
@@ -19,7 +21,18 @@ def show
...
+  end
+
+  def destroy
+    review = Review.find(params[:id])
+    park_id = review.amusement_park.id
+
+    if review.destroy
+      render json: { body: "deleted successfully", parkId: park_id }
```
Comment (bkincart) [inline app/controllers/api/v1/reviews_controller.rb]: snake_case

File: app/controllers/api/v1/amusement_parks_controller.rb [hunk 4]
```diff
@@ -74,4 +81,8 @@ def serialized_review
   def serialized_rides
     ActiveModel::Serializer::ArraySerializer.new(AmusementPark.find(params[:id]).rides, each_serializer: RideSerializer)
   end
+
+  def admin_status?
+    current_user.admin?
+  end
```
Comment (bkincart) [inline app/controllers/api/v1/amusement_parks_controller.rb]: I don't think you're using this method for this controller, please remove

File: spec/javascript/exampleTest.js [hunk 5]
```diff
@@ -1,127 +0,0 @@
...
-    it('render the park rides', (done) => {
-      setTimeout(() => {
-        expect(wrapper.text()).toMatch(ride[0].name)
-        done()
-
-      }, 0)
-    })
-
-  })
-});
```
Comment (bkincart) [inline spec/javascript/exampleTest.js]: Where did these tests end up going? Were they already duplicated in a different file?

Comment (Cheese-Baron) [reply spec/javascript/exampleTest.js]: yes it was duplicated in the other test

**Pull request merged**